distributed/scheduler.py

```diff
@@ -1554,6 +1556,15 @@ def get_nbytes(self, stream=None, keys=None, summary=True):
 
             return result
 
+    @property
+    def published_data_keys(self):
+        return set(self.published_data.keys())
+
+    def publish_data(self, keys=None, data=None, name=None):
+        print('puiblish!', keys, data, name)
+        self.published_data[name] = (data, keys)
```

Comment:
FWIW I recommend a dictionary here (dictionaries everywhere!) just to be more informative for other people coming to the code.